examples/gzip/gzip-envoy.yaml

```diff
@@ -0,0 +1,128 @@
+              - "*"
+              routes:
+              - match:
+                  prefix: "/stats/prometheus"
+                route:
+                  cluster: gzip
+          http_filters:
+          - name: envoy.filters.http.compressor
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.compressor.v3.Compressor
+              response_direction_config:
+                common_config:
+                  min_content_length: 100
+                  content_type:
+                    - text/plain
+                disable_on_etag_header: true
+              request_direction_config:
+                common_config:
+                  enabled:
+                    default_value: false
+                    runtime_key: request_compressor_enabled
+              compressor_library:
+                name: text_optimized
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip
+                  memory_level: 3
+                  window_bits: 10
+          - name: envoy.filters.http.router
+            typed_config: {}
+  - address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 8089
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          codec_type: AUTO
+          stat_prefix: ingress_http
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: backend
+              domains:
+              - "*"
+              routes:
+              - match:
+                  prefix: "/plain"
+                route:
+                  cluster: service
+              - match:
+                  prefix: "/json"
+                route:
+                  cluster: service
+          http_filters:
+          - name: envoy.filters.http.compressor
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.compressor.v3.Compressor
+              response_direction_config:
+                common_config:
+                  min_content_length: 100
+                  content_type:
+                    - application/json
+                disable_on_etag_header: true
+              request_direction_config:
+                common_config:
+                  enabled:
+                    default_value: false
+                    runtime_key: request_compressor_enabled
+              compressor_library:
+                name: text_optimized
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip
+                  memory_level: 3
+                  window_bits: 10
+          - name: envoy.filters.http.router
+            typed_config: {}
+
+  clusters:
+  - name: gzip
```

Comment:
i think rename this cluster gzip -> envoy-stats or similar - otherwise its confusing